### PR TITLE
Themes: 'Thanks' modal on activation from Theme Sheet

### PR DESCRIPTION
**Pull request by seear**
Acknowledge activation of a theme from the theme sheet with the existing 'Thanks' modal dialog.

screen shot 2016-05-06 at 13 55 06

Currently only works when a site is selected. A site selector for the sheet is coming soon.

**To Test**
* Go to http://calypso.localhost:3000/design and choose a site
* Click on the ... button for a theme and select __Details__
* Click the __Pick this design__ button
Expected: The modal should display
Expected: The first link in the modal should be to the customizer
* Check that the links in the modal work as expected

### CODE REVIEW

**Comment by ockham** [comment]
There already is an `EmptyComponent` in `test/test/helpers/react/empty-component` -- can we re-use that? (You could add your README to that directory tho.)

**Comment by seear** [comment]
> There already is an EmptyComponent in test/test/helpers/react/empty-component -- can we re-use that?

Yeah, good point, and I thought about this a bit at the time. In the end I wasn't totally happy with the tests and production code sharing the same empty component, since it seems they serve different purposes. Hopefully the test version will be longer lived than this production one.

File: client/my-sites/themes/thanks-modal.jsx [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable react/jsx-no-bind */
```
Comment (ockham) [inline client/my-sites/themes/thanks-modal.jsx]: Please don't unless there's a way more compelling reason than elsewhere :-)

What about changing the `trackClick` method as follows instead:

```js
boundTrackClick: function( eventName ) {
  return Helpers.trackClick.bind( null, 'current theme', eventName );
}
```

That should allow usage like

```js
onClick={ this.boundTrackClick( 'features' )}
```

**Comment by ockham** [comment]
> In the end I wasn't totally happy with the tests and production code sharing the same empty component, since it seems they serve different purposes. 

Yeah. :-/ Alternatively, tests could also use the production variant. But I guess I'm fine with the dupe for now since it's such a small component.

File: client/my-sites/theme/main.jsx [hunk 2]
```diff
@@ -216,10 +219,17 @@ const ThemeSheet = React.createClass( {
...
+		const siteID = this.props.selectedSite && this.props.selectedSite.ID;
 
 		return (
 			<Main className="themes__sheet">
 				{ this.renderBar() }
+				<ActivatingTheme siteId={ siteID }>
+					<ThanksModal
+						site={ this.props.selectedSite }
+						source={ 'details' }
+						clearActivated={ bindActionCreators( clearActivated, this.props.dispatch ) }/>
```
Comment (ockham) [inline client/my-sites/theme/main.jsx]: Consider using the second argument to the `connect` call, [`mapDispatchToProps`](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options), instead:

```js
export default connect(
	( state ) => {
		/* ... */
	},
	{ clearActivated }
)( ThemeSheet );
```

(and just use `clearActivated={ this.props.clearActivated }` then -- could do the same with the other actions you `dispatch`, BTW.)

**Comment by ockham** [comment]
Noting here that with premium themes, the flow works fine, but I end up at `/design/<mysite>` after purchase (instead of back at the theme sheet).

**Comment by ockham** [comment]
Left a couple of nitpicks. Looks good overall, and tests well!

**Comment by seear** [comment]
Regarding the `jsx-no-bind`:

> Please don't unless there's a way more compelling reason than elsewhere :-)

I disagree here, because this component will never be on a hot path, so it's not worth obfuscating code to prevent the eslint warning.

I have moved to your suggestion because the `event` param was preventing the _click_ verb being appended in helpers.js, but note that there is still a _bind_ for each prop—it's just that eslint is not aware of it!

**Comment by ockham** [comment]
:shipit:

**Pull request merged**